source/Octodiff/Core/BinaryDeltaWriter.cs

```diff
@@ -39,16 +39,15 @@ public void WriteDataCommand(Stream source, long offset, long length)
             {
                 source.Seek(offset, SeekOrigin.Begin);
 
-                var buffer = new byte[Math.Min((int)length, 1024 * 1024)];
+                var buffer = new byte[1024 * 1024];
 
                 int read;
-                long soFar = 0;
-                while ((read = source.Read(buffer, 0, (int)Math.Min(length - soFar, buffer.Length))) > 0)
+                do
                 {
-                    soFar += read;
+                    read = source.Read(buffer, 0, buffer.Length);
 
                     writer.Write(buffer, 0, read);
-                }
+                } while (read == buffer.Length);
```

Comment:
You should change the `read > buffer.length` to `read > 0` as the stream might return less then what you requested.

https://msdn.microsoft.com/en-us/library/system.io.filestream.read%28v=vs.110%29.aspx
 > An implementation is free to return fewer bytes than requested even if the end of the stream has not been reached.